src/mesh/boundary_standard.cxx

```diff
@@ -271,6 +271,8 @@ void BoundaryDirichlet::apply(Field2D &f,BoutReal t) {
           }	
         }
       }
+    } else {
+      throw BoutException("Unhandled staggering in BoundaryDirichlet.");
```

Comment:
Could we bring all these `throw`s to the top of the functions as explicit preconditions? I guess with `ASSERT1`, but `ASSERT0` might also be ok?

Reply:
I re-ordered the if-else statements. Mega whitespace changes: github does a terrible job with the diff, but if you pull the branch `git -w HEAD^^` does a much better job of ignoring the whitespace changes...